tests/components/brottsplatskartan/test_config_flow.py

```diff
@@ -0,0 +1,224 @@
+    RESULT_TYPE_CREATE_ENTRY,
+    RESULT_TYPE_FORM,
+)
+
+from tests.common import MockConfigEntry
+
+
+async def test_form(hass: HomeAssistant) -> None:
+    """Test we get the form."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.brottsplatskartan.config_flow.uuid.getnode",
+        return_value="1234567890",
+    ), patch(
+        "homeassistant.components.brottsplatskartan.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_AREA: "N/A",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "Brottsplatskartan HOME"
+    assert result2["data"] == {
+        "area": "N/A",
+        "latitude": None,
+        "longitude": None,
+        "app_id": "ha-1234567890",
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_location(hass: HomeAssistant) -> None:
+    """Test we get the form using location."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.brottsplatskartan.config_flow.uuid.getnode",
+        return_value="1234567890",
+    ), patch(
+        "homeassistant.components.brottsplatskartan.async_setup_entry",
+        return_value=True,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_AREA: "N/A",
+                CONF_LOCATION: {
+                    CONF_LATITUDE: 59.32,
+                    CONF_LONGITUDE: 18.06,
+                },
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "Brottsplatskartan 59.32, 18.06"
+    assert result2["data"] == {
+        "area": "N/A",
+        "latitude": 59.32,
+        "longitude": 18.06,
+        "app_id": "ha-1234567890",
+    }
+
+
+async def test_form_area(hass: HomeAssistant) -> None:
+    """Test we get the form using area."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.brottsplatskartan.config_flow.uuid.getnode",
+        return_value="1234567890",
+    ), patch(
+        "homeassistant.components.brottsplatskartan.async_setup_entry",
+        return_value=True,
+    ):
```

Comment:
Not needed if `pytestmark = pytest.mark.usefixtures("mock_setup_entry")` is used.